### DISCUSSION TITLE
FST language Model support for online transducer models?

### DISCUSSION THREAD

**Discussion by matt-roddy** [k2-fsa/sherpa-onnx#1166] [General]
Hi, I'm wondering whether there are any plans to support FST (N-gram) language models for use with online transducer models? I see we can use RNN LMs with transducers but I would prefer to use FSTs, if possible.

**Comment by csukuangfj** [comment]
Could you try our CTC models, which support HLG decoding?

**Comment by matt-roddy** [comment]
I have tried the CTC models with HLG decoding. I was just curious about using FST LMs with transducer models.  It sounds like there aren't any immediate plans for this. 
Thanks for the reply and all the great work on these repos!

**Reply by csukuangfj** [reply]
IIRC, we don't get a large improvement in WER using n-gram based LM with transducer models in icefall.

**Reply by matt-roddy** [reply]
Yeah, that makes sense. The use-case I was hoping to support though is for improving WER in different production contexts using different n-gram LMs trained on transcripts from the different contexts. This would avoid having to re-train/fine-tune the transducer model.
